api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -535,14 +535,13 @@ public void purgeUser(User user, boolean cascade) throws APIException {
 	 */
 	private void checkPrivileges(Role role) {
 		Collection<Privilege> privileges = role.getPrivileges();
-		
-		if (privileges != null) {
-			for (Privilege p : privileges) {
-				if (!Context.hasPrivilege(p.getPrivilege())) {
-					throw new APIAuthenticationException("Privilege required: " + p);
-				}
-			}
-		}
+
+		Optional.ofNullable(role.getPrivileges())
+				.map(p -> p.stream().filter(pr -> !Context.hasPrivilege(pr.getPrivilege())).collect(Collectors.toSet()))
+				.ifPresent(missing -> {
```

Comment:
I don't think you technically need the curly braces here since there's only one line of code. Makes it a bit sorter to read.

Reply:
Does that mean a modification of this? https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-CodeStyle

Reply:
I completely agree with that part of the coding conventions as it applies `if` blocks, `for` loops etc. These should always have curly braces.

However, we may need to have some guidance for lambdas, since I think the conventions there should be somewhat different, e.g., lambdas should not exceed a few lines and should not be wrapped in curly braces unless absolutely necessary. (This latter is to discourage lambdas for things that should be written as a separate method).

I'm more than willing to be contradicted on this, however.

Reply:
Thanks cc. @ibacher ,@dkayiwa @mks-d , as fur as i understand lambda,,@ibachers suggestion of reducing code by extending -> will be only remaining unless if otherwise